Add Alpine Dockerfile

michaeljs1990 opened the pull request:
This adds a docker file that can be used to make deploying the application
easier. It does not focus on trying to be an all in one solution as you
still need to setup mysql. The full image is only 55.5MB and makes a small
change to the config file that lets you use environment variables to set
config values all name spaced with IPAM_* for safety.

- [x] add documentation for image
- [x] test out functionality more to ensure everything is in working order

michaeljs1990 committed:
Add Alpine Dockerfile

This adds a docker file that can be used to make deploying the application
easier. It does not focus on trying to be an all in one solution as you
still need to setup mysql. The full image is only 55.5MB and makes a small
change to the config file that lets you use environment variables to set
config values all namespaced with IPAM_* for safty.

michaeljs1990 committed:
Docs and missing module

This adds some simple docs on how to get the container running as well as a missing
ctype package which i found throwing some errors.

michaeljs1990 commented on config.dist.php:
```diff
@@ -2,11 +2,11 @@
-$db['pass'] = "phpipamadmin";
-$db['name'] = "phpipam";
-$db['port'] = 3306;
+$db['host'] = getenv('IPAM_HOST') ? getenv('IPAM_HOST') : "localhost";
```
I used this instead of the shorthand one since it looks like you still are trying to support php 5.1.

michaeljs1990 commented on Dockerfile:
```diff
@@ -0,0 +1,29 @@
+  && ln -sf /dev/stdout /var/log/apache2/access.log \
+  && ln -sf /dev/stderr /var/log/apache2/error.log \
+  && mkdir -p /run/apache2 \
+  && sed -i 's/#LoadModule rewrite_module modules\/mod_rewrite.so/LoadModule rewrite_module modules\/mod_rewrite.so/' /etc/apache2/httpd.conf
```
No fancy mod2en in alpine linux

michaeljs1990 commented on Dockerfile:
```diff
@@ -0,0 +1,29 @@
+  && ln -s /phpipam/config.dist.php /phpipam/config.php \
+  && ln -sf /dev/stdout /var/log/apache2/access.log \
+  && ln -sf /dev/stderr /var/log/apache2/error.log \
+  && mkdir -p /run/apache2 \
```
Don't know why but you need to create this or apache dies on startup since it can't create the pid file in this directory.

michaeljs1990 commented on INSTALL.txt:
````diff
@@ -3,4 +3,15 @@
+```
+$ docker build -t phpipam .
+$ docker run -p 8888:80 -e IPAM_HOST='192.168.1.5' -e IPAM_PASS='password' -e IPAM_USER='root' phpipam
+```
````
You might want to move this all onto your site but it looks like you are using WP right now for hosting that.

michaeljs1990 commented on apache.conf:
```diff
@@ -0,0 +1,7 @@
+    AllowOverride All
+    Require all granted
+  </Directory>
+</VirtualHost>
```
Kept apache.conf simple since I would except most people host this behind an HTTPS terminating load balancer or are using it in an environment where they don't care if it's over HTTPS.

michaeljs1990 reviewed (commented):
Info for reviewer

michaeljs1990 commented:
The php ctype module is needed as well for running the application which is not listed in your install docs.

mrlesmithjr commented:
@michaeljs1990 I did this with Ansible, Alpine for Docker if you want to check it out as it might give you some guidance. Maybe, maybe not. Just thought I would share as it might be some help for you.

https://github.com/mrlesmithjr/docker-ansible-phpipam/tree/alpine

Specifically, what is going on in this https://github.com/mrlesmithjr/docker-ansible-phpipam/blob/alpine/config/ansible/playbook.yml

michaeljs1990 commented:
nice thanks, the crons and dependencies list is very helpful 👍

mrlesmithjr commented:
You got it man! Hope it helps some.

michaeljs1990 commented:
Would love to get this into the repository let me know if i can do any other work on this.

michaeljs1990 committed:
Add more notes from a night of pain

rebased and ran into a bunch of issues since submodules are now required and
no errors are visible in the UI when they fail to load in. For whatever reason
on initial install the password will not set I have tried 3 different times
something is going on that I will try to debug but i can see after i reset the password
an error about php mailer is puked out so likely something to do with that (it's about
an email not being valid).

Some work also needs to be done around the database install process to better support
docker out of the box. The automated install is super nice but only allowing access to
localhost through this method means you have to log into mysql and run commands. Adding
an additional checkbox to the process like "allow remote access" or otherwise would fix
this up. I could see some security concerns around this since it is opening up mysql to
the world however if you do not have your firewall setup or are not running your db in a
container where the port is hidden from the outside world.

michaeljs1990 committed:
This kills me

https://github.com/gliderlabs/docker-alpine/issues/253

michaeljs1990 committed:
Add issue

GaryAllan commented:
Thanks. I've added a Dockerfile but taken a different approach for the ENV variables. See #2425
